Rule based html validation against hydration (#2493)
MarkBind/markbind: docs/devGuide/design/serverSideRendering.md

If you are unsure what elements are allowed within other elements, or what constitutes invalid HTML in general, a good resource to reference would be the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span).
</box>

<box type="info" seamless>

> Modern browsers have robust inbuilt mechanisms to auto-correct common causes of hydration. Therefore, to avoid high false positive rate, markbind only has a minimalist rule-based validation against violations. The code are in the `core/src/utils/htmlValidationUtils.ts`. Please help extend it when more violations that certainly cause hydration on browsers are spotted.

Reviewer: ```suggestion
Modern browsers have robust inbuilt mechanisms to auto-correct common causes of hydration. Therefore, to avoid high false positive rate, Markbind has a minimalist rule-based validation against violations. The code for this is located in `core/src/utils/htmlValidationUtils.ts`. 

Please help extend it when violations that cause hydration on browsers are spotted.
```
